UpdateDependencies.ps1

```diff
@@ -0,0 +1,129 @@
+﻿#
+# Copyright (c) .NET Foundation and contributors. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+#
+
+function EnsureEnvironmentVariable([string]$envVarName)
+{
+    If ([Environment]::GetEnvironmentVariable($envVarName) -eq $null)
+    {
+        throw "Can't find environment variable '$envVarName'"
+    }
+}
+
+function SetEnvIfDefault([string]$envVarName, [string]$value)
+{
+    If ([Environment]::GetEnvironmentVariable($envVarName) -eq $null)
+    {
+        [Environment]::SetEnvironmentVariable($envVarName, $value)
+    }
+}
+
+EnsureEnvironmentVariable "GITHUB_USER"
+EnsureEnvironmentVariable "GITHUB_EMAIL"
+EnsureEnvironmentVariable "GITHUB_PASSWORD"
+
+SetEnvIfDefault 'COREFX_VERSION_URL' 'https://raw.githubusercontent.com/dotnet/versions/master/dotnet/corefx/master/Latest.txt'
+SetEnvIfDefault 'GITHUB_ORIGIN_OWNER' $env:GITHUB_USER
+SetEnvIfDefault 'GITHUB_UPSTREAM_OWNER' 'dotnet'
+SetEnvIfDefault 'GITHUB_PROJECT' 'corefx'
+SetEnvIfDefault 'GITHUB_UPSTREAM_BRANCH' 'master'
+
+
+# Updates the dir.props file with the latest CoreFX build number
+function UpdateValidDependencyVersionsFile
+{
+    $CoreFxLatestVersion = Invoke-WebRequest $env:COREFX_VERSION_URL -UseBasicParsing
+    $CoreFxLatestVersion = $CoreFxLatestVersion.ToString().Trim()
+
+    if (!$CoreFxLatestVersion)
+    {
+        Write-Error "Unable to find latest CoreFX version at $env:COREFX_VERSION_URL"
+        return $false
+    }
+
+    $DirPropsPath = "$PSScriptRoot\dir.props"
+    $xml = new-object xml
+    $xml.PreserveWhitespace = $true
+    $xml.Load($DirPropsPath)
+
+    $nsMgr = new-object Xml.XmlNamespaceManager $xml.NameTable
+    $nsMgr.AddNamespace('b', 'http://schemas.microsoft.com/developer/msbuild/2003')
+
+    $node = $xml.SelectSingleNode('//b:ValidationPattern[@Include = "CoreFxVersions"]/b:ExpectedPrerelease', $nsMgr)
+    $node.InnerText = $CoreFxLatestVersion
+
+    $xml.Save($dirPropsPath)
+    return $true
+}
+
+# Updates all the project.json files with out of date version numbers
+function RunUpdatePackageDependencyVersions
+{
+    cmd /c $PSScriptRoot\build.cmd /t:UpdateInvalidPackageVersions | Out-Host
+
+    return $LASTEXITCODE -eq 0
+}
+
+# Creates a Pull Request for the updated version numbers
+function CreatePullRequest
+{
+    git add . | Out-Host
+
+    $UserName = $env:GITHUB_USER
+    $Email = $env:GITHUB_EMAIL
+    $CommitMessage = 'Updating dependencies from latest build numbers'
```

Comment:
I think it might be useful to include the target version so it's easier to see in git history and shows up in the pull request title.